src/provider.c

```diff
@@ -1152,6 +1174,14 @@ int OSSL_provider_init(const OSSL_CORE_HANDLE *handle, const OSSL_DISPATCH *in,
         p11prov_ctx_free(ctx);
         return ret;
     }
+    /* Copy the returned module path if there was one */
+    if (ctx->module != NULL) {
```

Comment:
Wonder if it would be more clear to use a const char *modconf local variable, and then copy it to ctx->module.

I am afraid the current code may still lead to an attempt to free memory we shouldn;t when openssl tries to unload the provider after this fatal error.

Reply:
I am not a huge fan of const chars. In this case, it is not clear where it is stored and when it will stop existing. Is it bound to the openssl context? Provider context? Something else?

Lets use temporary variable, which should cover all the corner cases.